src/cljs/kixi/hecuba/tabs/hierarchy/xyplot.cljs

```diff
@@ -0,0 +1,354 @@
+            (om/update-state! owner #(assoc %
+                                            :sort-spec {:sort-key th-click
+                                                        :sort-asc true}))))
+        (recur)))
+    om/IRenderState
+    (render-state [_ {:keys [sort-spec]}]
+      (let [{:keys [sort-key sort-asc]} sort-spec
+            sensors (:sensors cursor)
+            sensors-data      (sensors/fetch-sensors (:property-details cursor) [])
+            sensors-with-data (filter #(every? seq [(:lower_ts %) (:upper_ts %)]) sensors-data)
+            table-id                    "sensors-table"]
+        (html
+         [:div.col-md-12 {:style {:overflow "auto"}}
+          [:table {:class "table table-hover table-condensed"}
+           [:thead
+            [:tr
+             (raw-data/sorting-th owner "X" :x)
+             (raw-data/sorting-th owner "Y" :y)
+             (raw-data/sorting-th owner "Description" :description)
+             (raw-data/sorting-th owner "Type" :type)
+             (raw-data/sorting-th owner "Unit" :unit)
+             (raw-data/sorting-th owner "Period" :period)
+             (raw-data/sorting-th owner "Resolution" :resolution)
+             (raw-data/sorting-th owner "Device ID" :device_id)
+             (raw-data/sorting-th owner "Location" :location)
+             (raw-data/sorting-th owner "Earliest Event" :lower_ts)
+             (raw-data/sorting-th owner "Last Event" :upper_ts)
+             (raw-data/sorting-th owner "Status" :status)]]
+           [:tbody
+            (for [row (if sort-asc
+                        (sort-by sort-key sensors-with-data)
+                        (reverse (sort-by sort-key sensors-with-data)))]
+              (om/build sensor-row {:sensor row :xyplot-data (:xyplot-data cursor)} {:opts {:event-chan event-chan}}))]]])))))
+
+(defn calc-linear-regression-line
+  [data]
+  (let [n (->> data count)
+        sum #(reduce + %)
+        sum-x (->> data (map first) sum)
+        sum-y (->> data (map second)  sum)
+        sum-x-times-y (->> data (map (fn [[x y]] (* x y))) sum)
+        sum-x-times-x (->> data (map (fn [[x y]] (* x x))) sum)
+        a (/ (- (* sum-y sum-x-times-x) (* sum-x sum-x-times-y))
+             (- (* n sum-x-times-x) (* sum-x sum-x)))
+        b (/ (- (* n sum-x-times-y) (* sum-x sum-y))
+             (- (* n sum-x-times-x) (* sum-x sum-x)))
+        fn #(float (+ a (* b %)))
+        min-x (->> data (map first) (apply min))
+        max-x (->> data (map first) (apply max))
+        xy1 [min-x (fn min-x)]
+        xy2 [max-x (fn max-x)]]
+    [xy1 xy2]))
+
+(defn process-data
+  [[x-data y-data]]
+  (let [x-raw-data (->> x-data (sort-by :timestamp) (mapv :value))
+        y-raw-data (->> y-data (sort-by :timestamp) (mapv :value))]
+    (if (not= (count x-raw-data) (count y-raw-data))
+      (throw (js/Error. "X and Y have different amounts of data (" (count x-raw-data) "/" (count y-raw-data) ")")))
```

Comment:
there might be gaps in just one of the datasets. Could we change this to filter out points where we are missing an x or a y rather than throwing an error?